### PR TITLE
Add lsp range info for file load errors 

### PR DESCRIPTION
**Pull request by faldor20**
This now makes the language server use the correct range when encountering a file load error.

So now parse errors won't all appear at the very top!

### CODE REVIEW

**Review by Anton-4 — DISMISSED** [review]
Looks good so far, just the test failures remaining

**Comment by faldor20** [comment]
I don't really understand that fuzzing test, or how I could have broken it.

**Comment by smores56** [comment]
@faldor20 seems like we'll be disabling the fuzzing step for now: https://roc.zulipchat.com/#narrow/channel/395097-compiler-development/topic/Weird.20fuzzing.20bug.20of.20the.20day/near/488547602

File: crates/compiler/load/build.rs
Comment (smores56) [inline crates/compiler/load/build.rs:91]: Why only display reports with no region?

File: crates/language_server/src/convert.rs
Comment (smores56) [inline crates/language_server/src/convert.rs:113]: This will presumably make the whole screen red if we don't have a region. Maybe just the first char is a better default? If not, then maybe we should use `usize::MAX` instead of "some really big number".

File: crates/language_server/src/convert.rs
Comment (smores56) [inline crates/language_server/src/convert.rs:117]: Presumably we either want to remove these `debug!`s or give more context to whoever would be reading them in the LSP logs.

**Comment by smores56** [comment]
There's nothing blocking merging this IMO, but I didn't approve since there's a CI check failing. Feel free to ping me when that's fixed and I can approve without worry of the approval being dismissed by code changes.

**Comment by smores56** [comment]
Thanks so much for this change! I started working on this at one point because it's really annoying when this happens and I end up needing to run `roc check` in another terminal to figure out how to fix the issue.

File: crates/language_server/src/convert.rs
Comment (faldor20) [reply crates/language_server/src/convert.rs:113]: USize max is a bad plan. Lsp and json goes to javascript which, we must remember stores numbers as 64bit float and that would overflow on a usezie from 64bit systems. 

This is stupid large and should be fine in all practical applications, but I guess I could use f64max.

We do this because the main cases where we don't have a target it's because the whole document is cooked in some way. 

As Anton mentioned in zulip, it's often hard to find these errors if they are just on line 1 or something. I agree. If the error is a blocker for even parsing which precludes any other errors, why not show the whole doc as an erorr

File: crates/compiler/load/build.rs
Comment (faldor20) [reply crates/compiler/load/build.rs:91]: Haha oops, good catch.

File: crates/language_server/src/convert.rs
Comment (faldor20) [reply crates/language_server/src/convert.rs:113]: But I guess we don't care right, because it won't actually overflow it'll just be imprecise.. I still don't entirely trust it to handle super large numbers maybe I'm just too conservative 😅

I guess I worry the imprecision will do funky things.

Comment (faldor20) [reply crates/language_server/src/convert.rs:113]: My arbitrary number should be about 10million chars is about 100k lines of code which is as much as anyone should ever need... 10k might be a little too small.

**Review by smores56 — APPROVED** [review]

**Comment by smores56** [comment]
Can you actually sign all of these commits @faldor20 ? https://github.com/roc-lang/roc/blob/main/CONTRIBUTING.md#commit-signing

**Comment by faldor20** [comment]
there you go @smores56  I think it lost my signatures because I got github to rebase onto master

**Pull request merged**
